tests/system_tests_qdstat.py

```diff
@@ -895,6 +921,29 @@ def test_ssl_trustfile_cert_to_unsecured(self):
         def test_ssl_bad_trustfile_to_unsecured(self):
             self.ssl_test_bad('unsecured_s', ['bad_trustfile'])
 
+        def test_ssl_peer_name_verify_disabled(self):
+            """
+            Verify the --ssl-disable-peer-name-verify option
+            """
+            params = [x for x in self.get_ssl_args()['trustfile']]
+            try:
+                self.run_qdstat(address="amqps://127.0.0.1:%s" % self.strict_port,
+                                args=['--general'] + params)
+                self.assertTrue(False, "Expected connection attempt to fail!")
+            except AssertionError:
+                # Expect the connection to fail, since the certificate has
+                # 'localhost' as the peer name and we used '127.0.0.1' instead.
+                pass
```

Comment:
Don't catch `AssertionError`? Catching AssertionError makes the test pass when the connection attempt does not fail.

Reply:
This was my main comment from looking it over. I expect its because run_qdstat performs asserts and throws on failure, per ssl_test_bad, but as you say catching that assertion will break the fallback failure when it succeeds connecting unexpectedly, since it obviously does too.